### ISSUE TITLE
Revise has trouble revising LinearAlgebra's structuredbroadcast.jl

### ISSUE THREAD

**Issue by mbauman** [timholy/Revise.jl#333]
First, my apologies for not reducing this down any further.  It looks like it has something to do with `::T{<:S}` short-form parametric argument declarations, but there may be more to it than that.  Here's a minimal reproducer on a 3-days old 1.3.0-alpha.148 (2019-08-16, commit b5f4e87774) with Revise v2.1.6:

```julia
julia> using Revise

julia> using LinearAlgebra

julia> Revise.track(LinearAlgebra)

julia> Diagonal(1:3) .+ Diagonal(1:3)
3×3 Diagonal{Int64,Array{Int64,1}}:
 2  ⋅  ⋅
 ⋅  4  ⋅
 ⋅  ⋅  6

shell> touch stdlib/LinearAlgebra/src/structuredbroadcast.jl

julia> Diagonal(1:3) .+ Diagonal(1:3)
┌ Warning: omitting call expression (UnionAll)(#s3<:Diagonal, LinearAlgebra.StructuredMatrixStyle{#s3<:Diagonal}) in ("none", 0)
└ @ Revise ~/.julia/packages/Revise/agmgx/src/lowered.jl:188
…
```

[Full log](https://gist.github.com/mbauman/f21ff6eb9155aca5353fd10359a9b758).

**Comment by timholy** [comment]
Can't reproduce with
```julia
(v1.4) pkg> st
    Status `~/.julia/environments/v1.4/Project.toml`
  [da1fd8a2] CodeTracking v0.5.7
  [aa1ae85d] JuliaInterpreter v0.7.0
  [6f1432cf] LoweredCodeUtils v0.3.7
  [bac558e1] OrderedCollections v1.1.0
  [295af30f] Revise v2.1.6 #master (https://github.com/timholy/Revise.jl.git)
```

**Comment by timholy** [comment]
Just submitted [2.1.7 to the registry](https://github.com/JuliaRegistries/General/pull/2812), once that merges perhaps try again.

**Comment by timholy** [comment]
2.1.7 is out now, so this should be easier to test.

**Comment by mbauman** [comment]
Fixed with Revise 2.1.7, thank you!
